### ISSUE TITLE
Type checks at strict but not at basic settings

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#3036]
**Describe the bug**

The following code type checks (no errors) at `strict` settings but fails type checking with `basic` settings.

**Expected behavior**

I would expect strict settings to be more strict and also fail. Here it looks like basic is more strict than strict.

**Screenshots or Code**

Minimal repro:

```python
from typing import Generic, TypeVar

_T = TypeVar("_T")

class Observer(Generic[_T]):
    def on_next(self, value: _T) -> None:
        ...

def default_if_empty(obv: Observer[_T], default_value: _T = None) -> None:
    obv.on_next(default_value)
```

**VS Code extension or command-line**

Both, version 1.1.221

**Comment by erictraut** [comment]
With `basic` type checking mode, `strictParameterNoneValue` is disabled, so the value of `default_value` is treated as `_T | None`. When `strictParameterNoneValue` is enabled, the value of `default_value` is treated as simply `_T`.

I think the correct fix here is to add special-case handling for function-scoped TypeVars when used as a parameter type annotation with a default argument expression. The resulting parameter type should be adjusted to include the type of the default value. I've added this special-case logic, and it will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.222, which I just published. It will also be included in the next release of pylance.
